### PR TITLE
modifying the multipath_test.py script for scenario testing

### PR DESCRIPTION
**Pull request by Naresh-ibm**
converted the code into functions, based on different scenarios,
So that each scenario can be ran when needed.
script also works fine as a whole.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
linux-7sxk:~ # avocado run multipath_test.py -m multipath_test.yaml
JOB ID     : fb7fff299cc9fd5cd87a5b2e762a811fbada3d14
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-18T03.03-fb7fff2/job.log
 (01/24) multipath_test.py:MultipathTest.test;queue-length-0674: PASS (2411.43 s)
 (02/24) multipath_test.py:MultipathTest.test;service-time-6a8f: PASS (2419.81 s)
 (03/24) multipath_test.py:MultipathTest.test;round-robin-da53: PASS (2419.86 s)
 (04/24) multipath_test.py:MultipathTest.test_fail_paths;queue-length-0674: PASS (2300.25 s)
 (05/24) multipath_test.py:MultipathTest.test_fail_paths;service-time-6a8f: PASS (2301.21 s)
 (06/24) multipath_test.py:MultipathTest.test_fail_paths;round-robin-da53: PASS (2300.30 s)
 (07/24) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;queue-length-0674: PASS (42.41 s)
 (08/24) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;service-time-6a8f: PASS (41.25 s)
 (09/24) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;round-robin-da53: PASS (42.44 s)
 (10/24) multipath_test.py:MultipathTest.test_io_run_on_single_path;queue-length-0674: PASS (388.76 s)
 (11/24) multipath_test.py:MultipathTest.test_io_run_on_single_path;service-time-6a8f: PASS (384.36 s)
 (12/24) multipath_test.py:MultipathTest.test_io_run_on_single_path;round-robin-da53: PASS (384.43 s)
 (13/24) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;queue-length-0674: PASS (398.05 s)
 (14/24) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;service-time-6a8f: PASS (398.24 s)
 (15/24) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;round-robin-da53: PASS (398.03 s)
 (16/24) multipath_test.py:MultipathTest.test_removing_all_paths;queue-length-0674: PASS (366.43 s)
 (17/24) multipath_test.py:MultipathTest.test_removing_all_paths;service-time-6a8f: PASS (366.68 s)
 (18/24) multipath_test.py:MultipathTest.test_removing_all_paths;round-robin-da53: PASS (365.18 s)
 (19/24) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;queue-length-0674: PASS (372.72 s)
 (20/24) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;service-time-6a8f: PASS (372.85 s)
 (21/24) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;round-robin-da53: PASS (372.83 s)
 (22/24) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;queue-length-0674: PASS (193.17 s)
 (23/24) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;service-time-6a8f: PASS (192.94 s)
 (24/24) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;round-robin-da53: PASS (192.96 s)
RESULTS    : PASS 24 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 19427.89 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-18T03.03-fb7fff2/results.html
linux-7sxk:~ #



linux-7sxk:~ # avocado run multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths -m multipath_test.yaml 
JOB ID     : 97126dfa72a1062185fcd0143a6dd84917053208
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-15T09.02-97126df/job.log
 (1/3) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;queue-length-0674: PASS (41.65 s)
 (2/3) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;service-time-6a8f: PASS (42.10 s)
 (3/3) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;round-robin-da53: PASS (41.24 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 125.19 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-15T09.02-97126df/results.html
linux-7sxk:~ #

linux-7sxk:~ # avocado run multipath_test.py:MultipathTest.test_io_run_on_single_path -m multipath_test.yaml
JOB ID     : fcd0b5dea13c5a579f8482d9e19df3bacec338fa
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-18T00.52-fcd0b5d/job.log
 (1/3) multipath_test.py:MultipathTest.test_io_run_on_single_path;queue-length-0674: PASS (389.05 s)
 (2/3) multipath_test.py:MultipathTest.test_io_run_on_single_path;service-time-6a8f: PASS (384.33 s)
 (3/3) multipath_test.py:MultipathTest.test_io_run_on_single_path;round-robin-da53: PASS (384.42 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1158.00 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-18T00.52-fcd0b5d/results.html
linux-7sxk:~ # 
linux-7sxk:~ # 
linux-7sxk:~ # 
linux-7sxk:~ # 
linux-7sxk:~ # avocado run multipath_test.py:MultipathTest.test_failing_reinstate_all_paths -m multipath_test.yaml
JOB ID     : d6586f34de2434ff360ed2c48fd4b7ff92f633db
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-18T01.13-d6586f3/job.log
 (1/3) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;queue-length-0674: PASS (398.18 s)
 (2/3) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;service-time-6a8f: PASS (398.27 s)
 (3/3) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;round-robin-da53: PASS (398.18 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1194.82 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-18T01.13-d6586f3/results.html
linux-7sxk:~ # 
linux-7sxk:~ # 
linux-7sxk:~ # avocado run multipath_test.py:MultipathTest.test_removing_all_paths -m multipath_test.yaml
JOB ID     : cf94fbefcde5659c453c5ea0856ab11fd0e12fc3
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-18T01.41-cf94fbe/job.log
 (1/3) multipath_test.py:MultipathTest.test_removing_all_paths;queue-length-0674: PASS (365.38 s)
 (2/3) multipath_test.py:MultipathTest.test_removing_all_paths;service-time-6a8f: PASS (365.35 s)
 (3/3) multipath_test.py:MultipathTest.test_removing_all_paths;round-robin-da53: PASS (366.47 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1097.40 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-18T01.41-cf94fbe/results.html
linux-7sxk:~ # 
linux-7sxk:~ # avocado run multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath -m multipath_test.yaml
JOB ID     : 7a72a30a33466931eb3132a6fc784a9481747be8
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-18T02.05-7a72a30/job.log
 (1/3) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;queue-length-0674: PASS (372.64 s)
 (2/3) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;service-time-6a8f: PASS (372.70 s)
 (3/3) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;round-robin-da53: PASS (372.84 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1118.38 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-18T02.05-7a72a30/results.html
linux-7sxk:~ # 
linux-7sxk:~ # avocado run multipath_test.py:MultipathTest.test_suspend_resume_all_mpath -m multipath_test.yaml
JOB ID     : 3be6323d1a2b228eb94a9780d898d5e86cb4cfab
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-11-18T02.39-3be6323/job.log
 (1/3) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;queue-length-0674: PASS (192.94 s)
 (2/3) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;service-time-6a8f: PASS (193.12 s)
 (3/3) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;round-robin-da53: PASS (193.00 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 579.25 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-11-18T02.39-3be6323/results.html
linux-7sxk:~ #

**Comment by Naresh-ibm** [comment]
[suspend_resume_all_paths_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3862729/suspend_resume_all_paths_job_log.txt)
[removing_adding_all_paths_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3862730/removing_adding_all_paths_job_log.txt)
[fail_reinstate_all_path_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3862731/fail_reinstate_all_path_job_log.txt)
[test_io_run_on_single_path_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3862732/test_io_run_on_single_path_job_log.txt)
[fail_reinstate_individual_paths.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3862733/fail_reinstate_individual_paths.txt)

File: io/disk/multipath_test.py [hunk 1]
```diff
@@ -151,46 +151,175 @@ def test(self):
                 if disk in multipath.get_paths(path_dic["wwid"]):
                     msg += "Blacklist of %s fails\n" % disk
             multipath.form_conf_mpath_file(defaults_extra=plcy)
+        if msg:
+            self.fail("Some tests failed. Find details below:\n%s", msg)
+
+        self.test_fail_paths()
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: why calling one test from another ?

File: io/disk/multipath_test.py [hunk 2]
```diff
@@ -151,46 +151,175 @@ def test(self):
...
+    def test_fail_paths(self):
+        '''
+        calling all function from here
+        '''
+        self.test_fail_reinstate_individual_paths()
+        self.test_io_run_on_single_path()
+        self.test_failing_reinstate_all_paths()
+        self.test_removing_all_paths()
+        self.test_suspend_resume_individual_mpath()
+        self.test_suspend_resume_all_mpath()
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py]: why calling one test from another ?

File: io/disk/multipath_test.py [hunk 3]
```diff
@@ -96,7 +96,7 @@ def setUp(self):
         for wwid in self.wwids:
             if wwid not in process.system_output('multipath -ll',
                                                  ignore_status=True,
-                                                 shell=True).decode("utf-8"):
+                                                 shell=True):
```
[the remark this answers is not in the document]
Comment (Naresh-ibm) [reply io/disk/multipath_test.py]: looks like I have missed this to remove.

File: io/disk/multipath_test.py (hunk 2 above)
Comment (Naresh-ibm) [reply io/disk/multipath_test.py]: when we run the complete test, we need these functions also to be run right?

Comment (narasimhan-v) [reply io/disk/multipath_test.py]: avocado (unittest) calls all `test_X()` by default. You need not call them.

Comment (Naresh-ibm) [reply io/disk/multipath_test.py]: ohhh ok, then will remove the function.
please let me know if you any more comments. I want to complete it on one shot

Comment (narasimhan-v) [reply io/disk/multipath_test.py]: @Naresh-ibm This  changes the flow of the test. Once you address this only we can look at the code in a meaningful way and review. Hope you understand that.

**Comment by Naresh-ibm** [comment]
#1432

**Pull request closed without merging**